### PR TITLE
DOC: signal: Make `_filter_design.py` plot labels consistent

### PR DESCRIPTION
**Pull request by oscargus**
#### What does this implement/fix?

* Consistent use and formatting of units in plot labels.
* Changes magnitude to amplitude (I personally prefer magnitude, but amplitude seems to be more frequent)
* Consistent use of Phase instead of Angle in plot labels
* Spell Nyquist with capital N in error messages
* Slightly easier to read example by introducing variables for cut-off and sample frequencies.

### CODE REVIEW

**Comment by oscargus** [comment]
I realize that this is just a subset of all signal functions. (Originally I was just about to change the `()` in freqz to `[]`...)

If this is the preferred format, I can continue to change in the rest of signal in later PRs.

File: scipy/signal/_filter_design.py [hunk 1]
```diff
@@ -5376,16 +5376,17 @@ def gammatone(freq, ftype, order=None, numtaps=None, fs=None):
     >>> from scipy import signal
     >>> import matplotlib.pyplot as plt
 
-    >>> b, a = signal.gammatone(440, 'iir', fs=16000)
+    >>> fc, fs = 440, 16000
```
Comment (AtsushiSakai) [inline scipy/signal/_filter_design.py:5379]: Should these two variables move to line 5367 and the 16 sampling example should also use these variables?

Comment (oscargus) [reply scipy/signal/_filter_design.py:5379]: I thought about it, but to me it looked like the idea was to keep the examples "separate". One can also remove the redundant imports on lines 5376 and 5377 and so on if that is preferred.

Just let me know what is preferred.

File: scipy/signal/_filter_design.py [hunk 2]
```diff
@@ -558,7 +558,7 @@ def freqz_zpk(z, p, k, worN=512, whole=False, fs=2*pi):
     >>> ax2 = ax1.twinx()
     >>> angles = np.unwrap(np.angle(h))
```
Comment (endolith) [inline scipy/signal/_filter_design.py:559]: Should these be changed to `phases` also?

**Comment by j-bowhay** [comment]
Friendly ping @oscargus

**Comment by melissawm** [comment]
Hi @oscargus :wave: can we help you move this forward? 😄

**Comment by oscargus** [comment]
Sorry about that. I'll try to read up, rebase, and have it done within a week or so.

**Comment by j-bowhay** [comment]
Friendly ping @oscargus

**Review by j-bowhay — APPROVED** [review]

**Pull request merged**